### PR TITLE
Organize functional Cypress tests to run in groups on Jenkins

### PR DESCRIPTION
**Pull request by cwdavies**
- Add a series of Cypress containers and config spec sets in docker-compose.e2e.yml
- Move cypress integration page tests into consumer-tools folder
- Disable screenshot capture when a test fails
- Install and try to use `docker-compose` instead of `docker run` in Jenkinsfile
- Fix admin cypress test

## How to test this PR

1. Verify that all cypress tests run in groups in docker containers
2. Check no groups of tests fail to complete due to timeout
3. Confirm that the admin test passes

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)

### CODE REVIEW

**Comment by willbarton** [comment]
Do we know what's up with that admin test that started failing after the last merge of main?

**Comment by cwdavies** [comment]
> Do we know what's up with that admin test that started failing after the last merge of main?

I will take a look at why the admin test has now just started failing and try to push a fix using this PR.

File: Jenkinsfile [hunk 1]
```diff
@@ -168,18 +171,55 @@ pipeline {
...
+                stage('Cypress rules-policy tests') {
+                    options {
+                        timeout(time: 10, unit: 'MINUTES')
+                    }
+                    steps {
+                        sh "./docker-compose -f docker-compose.e2e.yml run rules-policy-tests"
                     }
                 }
-
-                postGitHubStatus("jenkins/functional-tests", "success", "Passed", env.RUN_DISPLAY_URL)
```
Comment (willbarton) [inline Jenkinsfile:182]: Is there a reason we want to remove the posting of status back to the PR?

**Review by willbarton — COMMENTED** [review]
I have a question about the GitHub status reporting, and I don't really feel like I know enough to review the Cypress test changes.

File: Jenkinsfile (hunk 1 above)
Comment (cwdavies) [reply Jenkinsfile:182]: Good catch. I added status posting back to Jenkinsfile.

**Review by willbarton — APPROVED** [review]
Fantastic!

**Review by Scotchester — COMMENTED** [review]
Just recording a few comments we discussed while pairing on a review of this.

One additional thing: Please update the command for running the functional tests locally via the docker-compose.e2e.yml file.

File: Jenkinsfile [hunk 2]
```diff
@@ -161,25 +167,160 @@ pipeline {
...
+                CYPRESS_CMD = "npx cypress run -b chrome --headless"
+                CYPRESS_ENV = "-e CYPRESS_baseUrl=https://${env.CFGOV_HOSTNAME} -e CI=1"
+                CYPRESS_OPTIONS = "${CYPRESS_ENV} ${CYPRESS_SHM} ${CYPRESS_REPO} ${CYPRESS_CMD}"
+                HOST_UID_GID = sh(returnStdout: true, script: 'echo "$(id -u):$(id -g)"').trim()
+            }
+            parallel {
+                stage('admin-tests') {
+                    agent {
+                        label 'docker'
+                    }
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply Jenkinsfile]: Made this change to declare a single docker agent for all functional tests.

File: Jenkinsfile [hunk 3]
```diff
@@ -60,6 +61,12 @@ pipeline {
                     env.IMAGE_NAME_ES_LOCAL = "${env.IMAGE_ES_REPO}:${env.IMAGE_TAG}"
                 }
                 sh 'env | sort'
+                sh "curl -L https://github.com/docker/compose/releases/download/1.28.5/docker-compose-`uname -s`-`uname -m` -o docker-compose"
+                sh "chmod +x docker-compose"
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply Jenkinsfile]: Removed lines to install and remove docker-compose

File: Jenkinsfile [hunk 4]
```diff
@@ -161,25 +167,160 @@ pipeline {
...
+                stage('admin-tests') {
+                    agent {
+                        label 'docker'
+                    }
+                    options {
+                        timeout(time: 15, unit: 'MINUTES')
+                    }
+                    environment {
+                        CYPRESS_VOLUMES = "-v ${WORKSPACE}/test/cypress:/app1/test/cypress -v ${WORKSPACE}/cypress.json:/app1/cypress.json"
+                        DOCKER_CMD = "--rm ${CYPRESS_VOLUMES} -w /app1 ${CYPRESS_OPTIONS}"
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply Jenkinsfile]: That's a good suggestion to use the test stage name instead of `/app#`

**Pull request merged**